### ISSUE TITLE
script args order is broken after v3

### ISSUE THREAD

**Issue by ekesken** [argoproj/argo-workflows#8199]
## Summary

Breaking non-documented change causing failures during Argo v2 -> v3 switch due to order change in container args in Pod manifest for script templates.

in Argo v2 `script.args` values were reflecting to Pod manifest so:

```yaml
spec:
  containers:
  - args:
    - /argo/staging/script
    - some-value
    command:
    - python3
```

and in Argo v3, reflects like this:
```yaml
spec:
  containers:
  - args:
    - some-value
    - /argo/staging/script
    command:
    - python3
```

This order change causes an error like this at the end in Argo v3:
```
python3: can't open file 'some-value': [Errno 2] No such file or directory
````

We observed this problem during Argo v2.12.5 -> v3.1.5 upgrade, but I tested it in Argo v3.3.1 as well and hit the same issue.

## Diagnostics

The workflow that reproduces the problem:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: script-args-broken-in-v3.3.1
spec:
  entrypoint: main
  templates:
    - name: main
      dag:
        tasks:
          - name: test-script-call
            template: tmpl-script-with-args
            arguments:
              parameters:
                - name: some-param
                  value: "some-value"
    - name: tmpl-script-with-args
      inputs:
        parameters:
        - name: some-param
      script:
        image: python:3.8
        command: ["python3"]
        args:
         - "{{inputs.parameters.some-param}}"
        source: |
          import sys
          print(f"argv: '{sys.argv}'")
          print(f"stdin: '{sys.stdin.read()}'")
          print("some-param: '{{inputs.parameters.some-param}}'")
  
```

If you try it on v2, you'll see it succeeds and on v3 it will fail.

---

**Message from the maintainers**:

Impacted by this regression? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
v3 is 1yr old. I'm amazed this has not been raised before.

**Comment by alexec** [comment]
The ordering of args with script template is contraversial. Some in favor of prepend, some against.

@MrFreezeex @henrywangx @ivancili

The correct behaviour is the original behaviour. I don't know which of the two behaviours that is.

It maybe a regression to the correct original behaviour (i.e. wrong in v2.12).

**Comment by ivancili** [comment]
Current behavior is the original behavior (append the script name).
I introduced a breaking change by changing the logic to prepend the script name in [64ae330](https://github.com/argoproj/argo-workflows/commit/64ae33034d30a943dca71b0c5e4ebd97018448bf).
That change was reverted some time ago in [44a4f7e](https://github.com/argoproj/argo-workflows/commit/44a4f7e10ce1d88e82d5df86c000b93a422484e2).
It seems like @ekesken has been using a version with my breaking changes, before they were reverted.
